Hierarchies: Learning page for node labels' formatting (#665)
iTwin/presentation: packages/hierarchies/learning/Formatting.md

              nodeLabel: {
                selector: ECSql.createConcatenatedValueJsonSelector([
                  // create a selector for `CodeValue` property value
                  { propertyClassName: "BisCore.SpatialCategory", propertyClassAlias: "this", propertyName: "CodeValue" },
                  // include a static string value
                  { type: "String", value: " [" },
                  // create a selector for `ECInstanceId` property value in hex format
                  { selector: `printf('0x%x', ${ECSql.createRawPropertyValueSelector("this", "ECInstanceId")})` },
                  // include a static string value
                  { type: "String", value: "]" },
                ]),
              },
            })}
            FROM BisCore.SpatialCategory this
          `,
        },
      },
    ],
  },
});

// all returned nodes have their labels set in format "{CodeValue} [{ECInstanceId}]"
const nodes = hierarchyProvider.getNodes({ parentNode: undefined });
```

<!-- END EXTRACTION -->

Finally, in case 3, the grouping node's label is formatted automatically based on the type of grouping:

> - In case of class grouping, the labels is the class' display label and needs to extra formatting.

Reviewer: ```suggestion
- In case of class grouping, the label is the class' display label and needs no extra formatting.
```

- In case of label grouping, the label is taken from grouped nodes whose labels are already formatted.

> - In case of property grouping, the label is built using formatted property values. The formatter is applied to each property value separately using formatter used by `HierarchyProvider`.

Reviewer: ```suggestion
- In case of property grouping, the label is built using formatted property values. The formatter used by `HierarchyProvider` is applied to each property value separately`.
```
